### PR TITLE
Implement ``stable_silu_backward`` and update decomposition table

### PR DESCRIPTION
**Pull request by kavanase**
I found that my NequIP models were consistently immediately crashing in training when using extra-large model sizes, compile mode and OMol25 data. A lot of debugging and profiling located the ``silu_backward`` as the issue, which was overflowing on the first step – so no gradient clipping, LR reduction etc could fix. 

This is due to the expansion of the sigmoid into ``1 / (1 + exp(-x))`` within the decomposed ``silu_backward``, which causes overflows with large activations it seems, but using the ``torch.sigmoid`` primitive avoids this issue. Implemented here, checked that it trains fine and doesn't affect speeds.

### CODE REVIEW

**Comment by kavanase** [comment]
This issue and fix is somewhat related to this comment block below the `make_fx` function btw, so you may want to update/remove it:
https://github.com/mir-group/nequip/blob/0cb886607c20c22c17a4ab7703d8eba7a13b4ad2/nequip/utils/fx.py#L103-L112

**Review by cw-tan — APPROVED** [review]

**Comment by cw-tan** [comment]
Nice detective work @kavanase !
